server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -172,6 +173,7 @@ public static ThreadPoolType fromType(String type) {
         map.put(Names.SEARCH_THROTTLED, ThreadPoolType.RESIZABLE);
         map.put(Names.SYSTEM_READ, ThreadPoolType.FIXED);
         map.put(Names.SYSTEM_WRITE, ThreadPoolType.FIXED);
+        map.put(Names.REMOTE_STORE, ThreadPoolType.SCALING);
```

Comment:
I don't think we should re-use the snapshot thread pool because I think snapshot thread pool might be configured to more like offline/low priority executor while the remote store might be configured to be involved in online/higher priority traffic. Maybe to reuse other threadpool we can pick the search one ? I'm not sure either so what do you think ?

Reply:
Yeah, that makes sense. The SEARCH thread pool probably makes the most sense. It may well make sense to have a dedicated thread pool here, but I would want some benchmarking data to drive that decision. We will need to have the answer for this before promoting this feature out from behind the feature flag for now, but I don't think we can do the benchmarking until the caching is implemented, hence starting with SEARCH for now is probably the right decision.